tests/mat.rs

```diff
@@ -600,3 +624,79 @@ fn test_ortho() {
     assert!(na::approx_eq(&pm.znear(),  &24.0));
     assert!(na::approx_eq(&pm.zfar(),   &61.0));
 }
+
+#[test]
+fn test_cholesky_const() {
+    
+    let a : Mat3<f64> = Mat3::<f64>::new(1.0, 1.0, 1.0, 1.0, 2.0, 2.0, 1.0, 2.0, 3.0);
+    let g : Mat3<f64> = Mat3::<f64>::new(1.0, 0.0, 0.0, 1.0, 1.0, 0.0, 1.0, 1.0, 1.0);
+
+    let result = na::cholesky(&a);
+
+    match result {
+        Ok(v) => {
+                    assert!(na::approx_eq(&v, &g));
+
+                    let recomp = v * na::transpose(&v);
+                    assert!(na::approx_eq(&recomp, &a));
+                 },
+
+        Err(_) => assert!(false),
+    }
```

Comment:
I would just unwrap the result, as you expect it to be `Ok(…)` and then do the inner assertions. Just trying to avoid nesting to make the tests more readable. ;)